opennms-webapp/src/main/webapp/WEB-INF/jsp/graph/results.jsp

```diff
@@ -198,6 +199,29 @@
             </c:if>
         </h3>
      </div> <!-- panel-heading -->
+       <c:if test="${fn:contains(resultSet.resource.resourceType.label, 'SNMP Interface')}">
+         <c:if test="${fn:contains(resultSet.resource.label,'(*)') != true}">
+           <c:forEach var="attribute" items="${resultSet.resource.attributes}">
+             <c:if test="${fn:contains(attribute.name, 'ifIndex')}">
+               <c:set var="ifIndex" value="${attribute.value}"/>
+             </c:if>
+             <c:if test="${fn:contains(attribute.name, 'nodeId')}">
+               <c:set var="nodeId" value="${attribute.value}"/>
+             </c:if>
+           </c:forEach>
+           <%
+           String grafanaProtocol = System.getProperty("org.opennms.grafanaBox.protocol", "http");
+           String grafanaHostname = System.getProperty("org.opennms.grafanaBox.hostname", "localhost");
+           int grafanaPort = Integer.parseInt(System.getProperty("org.opennms.grafanaBox.port", "3000"));
+
+           %>
+           <div ng-app="onms-ksc" ng-controller="checkFlowsCtrl" ng-init="getFlowCount(${nodeId}, ${ifIndex}, ${results.start.time}, ${results.end.time})">
+               <div ng-show="hasFlows">
+                 <a href="<%=grafanaProtocol%>://<%=grafanaHostname%>:<%=grafanaPort%>/dashboard/flows?orgId=1&amp;var-node=${nodeId}&amp;var-interface=${ifIndex}" style="padding-right: 3px" title="Open Flows for ${resultSet.resource.name}"><button type="button" class="btn btn-default btn-xs"><i class="fa fa-bar-chart" aria-hidden="true"></i></span></button></a>
```

Comment:
We make a lot of assumptions here in the URL. I think we'd be better of to provide a URL template with placeholders and replace the nodeId, and ifIndex in the string appropriately.